sdk/storage_blobs/src/blob/operations/append_block.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{blob::operations::put_block::PutBlockResponse, prelude::*};
+use azure_core::prelude::*;
+use bytes::Bytes;
+
+#[derive(Debug, Clone)]
+pub struct AppendBlockBuilder {
+    blob_client: BlobClient,
+    body: Bytes,
+    hash: Option<Hash>,
+    condition_max_size: Option<ConditionMaxSize>,
+    condition_append_position: Option<ConditionAppendPosition>,
+    lease_id: Option<LeaseId>,
+    client_request_id: Option<ClientRequestId>,
+    timeout: Option<Timeout>,
+}
+
+impl AppendBlockBuilder {
+    pub(crate) fn new(blob_client: BlobClient, body: impl Into<Bytes>) -> Self {
```

Comment:
We normally keep these constructors non-generic and instead do the conversion inside the client operation method.